lib/akamai_rspec/matchers/x_cache_headers.rb

```diff
@@ -10,10 +10,12 @@ def x_cache_headers
   end
 end
 
-RSpec::Matchers.define :be_served_from_origin do |contents|
+RSpec::Matchers.define :be_served_from_origin do |*expected_with_custom_headers|
   include AkamaiRSpec::Helpers
   match do |url|
-    response = AkamaiRSpec::Request.get url
+    custom_headers = expected_with_custom_headers.size==2? expected_with_custom_headers[1]: {}
+    contents = expected_with_custom_headers[0]
```

Comment:
You mentioned that you want the option to pass a response in to here. To do that, could you please write a test for passing a response in and rename `url` to `url_or_response`?